src/scores/categorical/binary_impl.py

```diff
@@ -0,0 +1,149 @@
+    obs: xr.DataArray,
+    reduce_dims: Optional[Sequence[str]] = None,
+    preserve_dims: Optional[Sequence[str]] = None,
+    weights: Optional[xr.DataArray] = None,
+    check_args: Optional[bool] = True,
+) -> xr.DataArray:
+    """
+    Calculates the Probability of Detection (POD), also known as the Hit Rate.
+    This is the proportion of observed events (obs = 1) that were correctly
+    forecast as an event (fcst = 1).
+
+    Args:
+        fcst: An array containing binary values in the set {0, 1, np.nan}
+        obs: An array containing binary values in the set {0, 1, np.nan}
+        reduce_dims: Optionally specify which dimensions to sum when
+            calculating the POD. All other dimensions will be not summed. As a
+            special case, 'all' will allow all dimensions to be summed. Only one
+            of `reduce_dims` and `preserve_dims` can be supplied. The default behaviour
+            if neither are supplied is to sum across all dims.
+        preserve_dims: Optionally specify which dimensions to not sum
+            when calculating the POD. All other dimensions will be summed.
+            As a special case, 'all' will allow all dimensions to be
+            not summed. In this case, the result will be in the same
+            shape/dimensionality as the forecast, and the errors will be
+            the POD score at each point (i.e. single-value comparison
+            against observed), and the forecast and observed dimensions
+            must match precisely. Only one of `reduce_dims` and `preserve_dims` can be
+            supplied. The default behaviour if neither are supplied is to reduce all dims.
+        weights: Optionally provide an array for weighted averaging (e.g. by area, by latitude,
+            by population, custom)
+        check_args: Checks if `fcst and `obs` data only contains values in the set
+            {0, 1, np.nan}. You may want to skip this check if you are sure about your
+            input data and want to improve the performance when working with dask.
+
+    Returns:
+        A DataArray of the Probability of Detection.
+
+    Raises:
+        ValueError: if there are values in `fcst` and `obs` that are not in the
+            set {0, 1, np.nan} and `check_args` is true.
+
+    """
+    # fcst & obs must be 0s and 1s
+    if check_args:
+        check_binary(fcst, "fcst")
+        check_binary(obs, "obs")
+    dims_to_sum = gather_dimensions(fcst.dims, obs.dims, reduce_dims, preserve_dims)
+
+    misses = (obs == 1) & (fcst == 0)
+    hits = (obs == 1) & (fcst == 1)
+
+    # preserve NaNs
+    misses = misses.where((~np.isnan(fcst)) & (~np.isnan(obs)))
+    hits = hits.where((~np.isnan(fcst)) & (~np.isnan(obs)))
+
+    misses = apply_weights(misses, weights)
+    hits = apply_weights(hits, weights)
+
+    misses = misses.sum(dim=dims_to_sum)
+    hits = hits.sum(dim=dims_to_sum)
+
+    pod = hits / (hits + misses)
+
+    pod.name = "ctable_probability_of_detection"
```

Comment:
Why is the name prefixed with 'ctable', and is that meaningful for users?

Reply:
Left over from Jive - ctable refers to contingency table. I've removed this.